frontend/src/utils/scroll-to-top.tsx

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+import { StaticContext } from '../../node_modules/@types/react-router';
+
+
+class ScrollToTop extends React.Component<RouteComponentProps<any, StaticContext>, {}> {
+  componentDidUpdate(prevProps: RouteComponentProps<any, StaticContext>) {
+    if (this.props.location !== prevProps.location) {
+      setTimeout(() => window.scrollTo(0, 0), 0);
```

Comment:
this would not put scrollTo on top of the execution stack, but will queue it in the event loop which will give the chance for the browser to render UI (especially when clicking back button) and then, once rendered, we can scroll to top cc @mstockerl